### PR TITLE
BDS decoding

### PR DESCRIPTION
**Pull request by ibenacha**

### CODE REVIEW

File: src/main/java/de/serosystems/lib1090/msgs/modes/ACASActiveResolutionAdvisorReport.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+import de.serosystems.lib1090.exceptions.UnspecifiedFormatError;
+import de.serosystems.lib1090.msgs.ModeSDownlinkMsg;
+
+import java.io.Serializable;
+import java.util.Arrays;
+
+/**
+ * BDS 3,0
+ */
+public class ACASActiveResolutionAdvisorReport extends ModeSDownlinkMsg implements Serializable {
```
Comment (affitz) [inline src/main/java/de/serosystems/lib1090/msgs/modes/ACASActiveResolutionAdvisorReport.java:13]: There is a "y" missing in the class name.

**Review by affitz — CHANGES_REQUESTED** [review]
The overall structuring of the decoder within the libraries should be changed:

These decoders should not inherit from ModeSDownlinkMsg. BDS registers are not downlink messages, they are parts of two downlink messages, namely the two CommB downlink replies. The BDS decoders should only deal with decoding the [BDS data](https://github.com/SeRoSystems/lib1090/blob/master/src/main/java/de/serosystems/lib1090/msgs/modes/CommBAltitudeReply.java#L211). It should not be mixed with decoding the rest of the downlink replies which is already done in the existing CommB reply classes.

As a result of your design, you have a lot of redundant code that is identical in each of your classes below and is in addition redundant to what is already being done in the CommB reply decoders.

This needs to be restructured and the redundancy needs to be removed. I recommend to create a new interface or class called de.serosystems.lib1090.bds.BDSRegister and then each BDS decoder inherits from that.

**Review by affitz — CHANGES_REQUESTED** [review]
Thanks, looks much better now. Could you also add the respective Java doc? See the other lib1090 message classes. Best way to do it is to copy the descriptions of the fields from the Mode S spec. Cheers!

**Comment by ibenacha** [comment]
I added some doc based on the document in my hands (ED-73E Mode S MOPS Vol 2). I will sync with Jorg regarding some reserved fields. for the time being, let me know if the existing doc match your expectation. Thanks.

**Review by affitz — APPROVED** [review]
Looks good, thanks!

**Pull request merged**